src/databricks/labs/remorph/reconcile/connectors/snowflake.py

```diff
@@ -33,36 +50,28 @@ def read_data(self, catalog: str, schema: str, query: str, options: JdbcReaderOp
                     .load()
                 )
             return df.select([col(column).alias(column.lower()) for column in df.columns])
-        except PySparkException as e:
-            error_msg = (
-                f"An error occurred while fetching Snowflake Data using the following {query} in "
-                f"SnowflakeDataSource : {e!s}"
-            )
-            raise PySparkException(error_msg) from e
+        except RuntimeError as e:
+            raise self._raise_runtime_exception(e, "data", table_query)
 
     def get_schema(self, catalog: str, schema: str, table: str) -> list[Schema]:
+        schema_query = self.get_schema_query(catalog, schema, table)
         try:
-            schema_query = self.get_schema_query(catalog, schema, table)
             schema_df = self.reader(schema_query).load()
```

Comment:
Here we are calling `load` on the `DataFrameReader`.

Reply:
done